src/main/java/org/mastodon/mamut/nearest/NearestObjectStatFeature.java

```diff
@@ -0,0 +1,291 @@
+
+	private final Map< FeatureProjectionKey, MyDoublePropertyProjection > featureProjections;
+
+	private final Map< NearestObjectStatItem, FeatureProjectionKey > fSpecMap;
+
+	private NearestObjectStatFeature(
+			final Map< FeatureProjectionKey, MyDoublePropertyProjection > featureProjections,
+			final Map< NearestObjectStatItem, FeatureProjectionKey > fSpecMap )
+	{
+		this.featureProjections = featureProjections;
+		this.fSpecMap = fSpecMap;
+	}
+
+	/**
+	 * Returns the projection corresponding to the specified stat item, or
+	 * <code>null</code> if the results for the stat item are not stored in this
+	 * feature.
+	 * 
+	 * @param item
+	 *            the stat item.
+	 * @return the projection, or <code>null</code>.
+	 */
+	public FeatureProjection< Spot > get( final NearestObjectStatItem item )
+	{
+		final FeatureProjectionKey key = fSpecMap.get( item );
+		if ( key == null )
+			return null;
+		return featureProjections.get( key );
+	}
+
+	@Override
+	public FeatureProjection< Spot > project( final FeatureProjectionKey key )
+	{
+		return featureProjections.get( key );
+	}
+
+	@Override
+	public Set< FeatureProjection< Spot > > projections()
+	{
+		return new LinkedHashSet<>( featureProjections.values() );
+	}
+
+	@Override
+	public FeatureSpec< NearestObjectStatFeature, Spot > getSpec()
+	{
+		final Set< FeatureProjectionKey > set = featureProjections.keySet();
+		final FeatureProjectionSpec[] fps = new FeatureProjectionSpec[set.size()];
+		final Iterator< FeatureProjectionKey > it = set.iterator();
+		for ( int i = 0; i < set.size(); i++ )
+			fps[ i ] = it.next().getSpec();
+		return new MyFeatureSpec( fps  );
+	}
+
+	@Override
+	public void invalidate( final Spot obj )
+	{
+		for ( final MyDoublePropertyProjection fp : featureProjections.values() )
+			fp.invalidate( obj );
+	}
+
+	public void set( final Spot spot, final NearestObjectStatItem item, final double value )
+	{
+		final FeatureProjectionKey key = fSpecMap.get( item );
+		if ( key == null )
+			return;
+		final MyDoublePropertyProjection fp = featureProjections.get( key );
+		if ( fp == null )
+			return;
+
+		fp.set( spot, value );
+	}
+
+	public static NearestObjectStatFeature createFeature( final NearestObjectStatModel statModel, final Model model )
+	{
+		final Map< FeatureProjectionKey, MyDoublePropertyProjection > featureProjections = new LinkedHashMap<>( statModel.size() );
+		final Map< NearestObjectStatItem, FeatureProjectionKey > fSpecMap = new HashMap<>();
+
+		for ( final NearestObjectStatItem item : statModel )
+		{
+			Dimension projectionDimension;
+			switch ( item.value )
+			{
+			case DISTANCE_OR_N:
+			{
+				projectionDimension = ( item.collectBy == CollectBy.SPECIFY_N ) ? Dimension.LENGTH : Dimension.NONE;
+				break;
+			}
+			case FEATURE:
+			{
+				final FeatureProjectionSpec projection = getProjectionSpec( item.featureID, model.getFeatureModel() );
+				projectionDimension = projection.projectionDimension;
```

Comment:
The method `getProjectionSpec` may return `null`, but this case is not handled here. So there could be a NullPointerException here.